client/data/timeline/selectors.ts

```diff
@@ -0,0 +1,21 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { TimelineState } from './reducer';
+import { Timeline } from './types';
+
+export const getTimeline = (
+	state: { timeline: TimelineState },
+	id: string
+): Timeline | undefined => {
+	return state.timeline?.data?.[ id ];
+};
+
+export const getTimelineError = (
+	state: { timeline: TimelineState },
+	id: string
+): Error | undefined => {
```

Comment:
This changes the signature of the result a bit. Previously we'd return empty object if error is missing. I haven't tested site errors, so not 100% sure if it affects the UI.

Reply:
I chose to go with `undefined` instead of empty object because the type signature of an empty object is ugly; `Record<string, never>`.

I think it will be better for us to use `undefined` here as well because it's clearer what's going on; there's no error, not an empty error.

I do agree that we have to be careful here though, but I'm not sure how to test the error handling in the UI. Do I have to just change the code to return malformed data form the server?

Reply:
I like the change but it should be covered with corresponding test IMO.

> I do agree that we have to be careful here though, at least until everything has been migrated, but I'm not sure how to test the error handling in the UI. Do I have to just change the code to return malformed data form the server?

Yep, either throwing an error on server. Or if working in test set mock to throw error.

Reply:
> it should be covered with a corresponding test

[It already is](https://github.com/Automattic/woocommerce-payments/blob/6d0a079b71a903357d47f69c4238258dd4bc8281/client/data/timeline/test/selectors.ts#L50-L60) 🙂